src/pages/MainPage.js

```diff
@@ -0,0 +1,59 @@
+import { html } from 'htm/preact';
+import { Component } from 'preact';
+import { Totals } from '../components/Totals.js';
+import { WorldMap } from '../components/WorldMap.js';
+import { CountryInfo } from '../components/CountryInfo.js';
+import { router } from '../router.js';
+
+export class MainPage extends Component {
+  constructor() {
+    super();
+
+    this.__onPathChanged = this.__onPathChanged.bind(this);
+    this.__closeCountryInfo = this.__closeCountryInfo.bind(this);
+  }
+
+  async componentDidMount() {
+    const countriesData = await (await fetch(new URL('../../data/datafile.json', import.meta.url))).json();
+    this.setState({ countriesData });
+  }
+
+  componentWillMount() {
+    router.addEventListener('path-changed', this.__onPathChanged);
+  }
+
+  componentWillUnmount() {
+    router.removeEventListener('path-changed', this.__onPathChanged);
+  }
+
+  render() {
+    if (!this.state.countriesData) {
+      // Loading state here
+      return html``;
+    }
+
+    return html`
+      <${Totals} />
+      <${WorldMap} countriesData=${this.state.countriesData} />
```

Comment:
small tip, in preact you get the props and state passed in `render`,so:

`render(props, state) {}`